### ISSUE TITLE
DataMember Names / Constant Strings missing from build output.

### ISSUE THREAD

**Issue by HeathJared** [obfuscar/obfuscar#223]
We have a large application with tons of POCO class decorated with `[DataContract]` and `[DataMember]` attributes. When obfuscating, we regularly encounter `[DataMember]` names and other constant strings (log messages, variables, etc) that are randomly missing values in the build output. We have also encountered the issue with constant string log messages.

Obfuscar 2.2.23.0 (latest at the time) was install via NuGet Package.

The project is obfuscated with the following flags:

```
  <Var name="KeepPublicApi" value="false" />
  <Var name="ReuseNames" value="false" />
  <Var name="RenameProperties" value="true" />
  <Var name="RenameFields" value="true" />
  <Var name="HideStrings" value="true" />
  <Var name="SuppressIldasm" value="true" />  
```

Here is a simplified example of the issue (some properties removed):

Original Class

```
[DataContract]
public class TokenResponse
{
    [DataMember(Name = "access_token")]
    public string AccessToken { get; set; }

    [DataMember(Name = "token_type")]
    public string TokenType { get; set; }
}
```

Produced Output:

```
[DataContract]
public class bq
{
    [CompilerGenerated]
    private string Hh;

    [CompilerGenerated]
    private DateTimeOffset HK;

    [DataMember(Name = "access_token")]
    public string aEU
    {
        get;
        set;
    }

    [DataMember(Name = "")]
    public string aEX
    {
        get;
        set;
    }
}
```

Produced Map:

```
[assembly][namespace].TokenResponse -> [assembly]b.bq
{
	[assembly][namespace].TokenResponse::get_AccessToken[0]( ) -> A
	[assembly][namespace].TokenResponse::set_AccessToken[1]( [System.Runtime]System.String ) -> A
	[assembly][namespace].TokenResponse::get_TokenType[0]( ) -> D
	[assembly][namespace].TokenResponse::set_TokenType[1]( [System.Runtime]System.String ) -> B

	[assembly][namespace].TokenResponse::.ctor[0]( ) skipped:  special name

	System.String [assembly]System.String [namespace].TokenResponse::<AccessToken>k__BackingField -> Hh
	System.String [assembly]System.String [namespace].TokenResponse::<TokenType>k__BackingField -> Hk

	System.String [assembly]System.String [namespace].TokenResponse::AccessToken -> aEU
	System.String [assembly]System.String [namespace].TokenResponse::TokenType -> aEX
}
```

The decompiled code was produced using the ILSPY tool.

As you can see, the `aEX` / `TokenType` property is missing the `[DataMember]` name and our serialization and serialization process fails.

**Comment by lextm** [comment]
Can you run release 2.2.19? That should tell if this is a known issue.

**Comment by HeathJared** [comment]
Yes, we can reproduce the issue on 2.2.19.

One of my developers @jascen noticed that updating Mono.Cecil, from 0.10.3 to 0.10.4 (from NuGet) and rebuilding Obfuscar appears to have fixed the issue.

**Comment by lextm** [comment]
Thanks. That sounds like an upstream bug. I will investigate further from here.

**Comment by lextm** [comment]
I couldn't reproduce the issue.

Anyway, update Mono Cecil version and shipped 2.2.24.

Close it now.

**Comment by HeathJared** [comment]
We have not seen the issue since the Mono.Cecil update (over 30+) builds.

Thanks for the update.
